File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode addFirst(ListNode head, int val)
    {
        ListNode temp = new ListNode(val);
        temp.next = head;
        return temp;
    }
    public ListNode reverseList(ListNode head) {
        if(head == null)
            return head;
        if(head.next == null)
            return head;
        ListNode cur = head;
        ListNode ans = new ListNode(head.val);
        cur = cur.next;
        while(cur!=null)
        {
            ans = addFirst(ans, cur.val);
            cur = cur.next;
        }
        return ans;
    }
}

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        d = {}
        cur = head
        while cur != None and cur.next != None:
            i = cur.val
            n = cur.next
            j = n.val
            if i in d:
                if d[i] == j:
                    return True
                else:
                    continue
            else:
                d[i] = j
            cur = cur.next
        if cur == None or cur.next == None:
            return False
        return True